### ISSUE TITLE
ANN001 is fired when argument already set (`mypy` not complain in such case).

### ISSUE THREAD

**Issue by actionless** [charliermarsh/ruff#794]
```python
def function(param=False) -> int:
    if param:
        return 1
    return 0
```

```console
$ ruff test_ruff.py
Found 1 error(s).
test_ruff.py:1:14: ANN001 Missing type annotation for function argument `param`

$ mypy test_ruff.py
Success: no issues found in 1 source file
```

**Comment by edgarrmondragon** [comment]
FWIW `mypy` does catch the missing annotation in strict mode:

```console
$ mypy checkANN001.py --strict
checkANN001.py:1: error: Function is missing a type annotation for one or more arguments  [no-untyped-def]
Found 1 error in 1 file (checked 1 source file)
```

**Comment by actionless** [comment]
thanks for clarifying

i wonder, if it would make sense having such option as a separate check? (in a similar way how self or cls annotation checks were extracted out)

**Comment by charliermarsh** [comment]
It could make sense to turn this into a configuration setting, like `allow_star_arg_any` and `mypy_init_return`.

**Comment by actionless** [comment]
closing because it unlikely to be implemented in mypy:
https://stackoverflow.com/a/64500210/1850190
